src/list.js

```diff
@@ -61,6 +61,28 @@ export default class List extends Plugin {
 			}
 		} );
 
+		// Overwrite default Backspace key behavior.
+		// If Backspace key is pressed with selection collapsed on first position in first list item, outdent it.
+		this.listenTo( this.editor.editing.view, 'delete', ( evt, data ) => {
```

Comment:
As for this callback – I propose to refactor it a bit. Instead of one big `if()` let's have early returns. Why? The reason is performance aspect. If we have to access so many properties and calculate so many values on every <kbd>Backspace</kbd> we're increasing memory pressure and risking glitches.

Another thing is that we need to include two more cases here – when a list is nested (in a blockQuote e.g.) and when the block before is nested in a blockQuote.

Like:

```html
<bQ><p>x</p></bQ><listItem>[]y</listItem>

<p>x</p><bQ><listItem>[]y</listItem></bQ>
```

Please write tests for them. I think those cases will work fine with the current code, but just to be sure.